beep-beep/jobs.md

```diff
@@ -0,0 +1,695 @@
+
+For example, in this way, you can configure both the client and server parts to
+work with RabbitMQ.
+
+```yaml
+#
+# RPC is required for tasks dispatching (client)
+#
+rpc:
+  listen: tcp://127.0.0.1:6001
+
+#
+# This section configures the task consumer (server)
+#
+server:
+  command: php consumer.php
+  relay: pipes
+
+#
+# AMQP connection settings (for example, RabbitMQ)
+#
+amqp:
+    addr: amqp://guest:guest@localhost:5672
+
+#
+# In this section, the jobs themselves are configured
+#
+jobs:
+  consume: [ "test" ]   # List of RoadRunner queues that can be processed by 
+                        # the consumer specified in the "server" section.
+  pipelines:
+    test:               # RoadRunner queue identifier
+      driver: amqp      # - Queue driver name
+      queue: test       # - Internal (driver's) queue identifier
+```
+
+- The `rpc` section is responsible for client settings. It is at this address
+  that we will connect, *dispatching tasks* to the queue.
+
+- The `server` section is responsible for configuring the server. Previously, we
+  have already met with its description when setting up the [PHP Worker](/php/worker.md).
+
+- The `amqp` section contains settings for connecting to RabbitMQ. As the name
+  implies, the settings are specific only for the RabbitMQ driver and *may not*
+  be available if you use some other driver.
+
+- And finally, the `jobs` section is responsible for the work of the queues
+  themselves. It contains information on how the RoadRunner should work with
+  connections to drivers, what can be handled by the consumer, and other
+  queue-specific settings.
+
+### Common Configuration
+
+Let's now focus on the common settings of the queue server. In full, it may 
+look like this:
+
+```yaml
+jobs:
+  num_pollers: 64
+  pipeline_size: 100000
+  pool:
+    num_workers: 10
+    max_jobs: 0
+    allocate_timeout: 60s
+    destroy_timeout: 60s
+  consume: [ "queue-name" ]
+  pipelines:
+    queue-name:
+      driver: # "[DRIVER_NAME]"
+      # And driver-specific configuration below...
+```
+
+Above is a complete list of all possible common Jobs settings. Let's now figure
+out what they are responsible for.
+
+- `num_pollers` - TODO
+- `pipeline_size` - TODO
+- `pool.num_workers` - TODO
+- `pool.max_jobs` - TODO
+- `pool.allocate_timeout` - TODO
+- `pool.destroy_timeout` - TODO
+- `consume` - Contains an array of the names of all queues specified in the
+  `"pipelines"` section, which should be processed by the concierge specified in
+   the global `"server"` section (see the [PHP worker's settings](/php/worker.md)).
+- `pipelines` - This section contains a list of all queues declared in the
+  RoadRunner. The key is a unique *queue identifier*, and the value is an object
+  from the settings specific to each driver (we will talk about it later).
+
+### Ephemeral (Memory) Driver
+
+TODO
+
+### AMQP Driver
+
+TODO
+
+### Beanstalk Driver
+
+TODO
+
```

Comment:
```json
      driver: beanstalk
      priority: 11
      tube_priority: 1
      tube: default-1
      reserve_timeout: 10s
```
Where:
1. `driver` - driver name.
2. `priority` - same as the amqp. Default: 10.
3. `tube_priority` - there are two priorities for the beanstalk driver. First - RR pipeline priority with all jobs from all available drivers. Second - internal beanstalk tube priority which can be set separately. Default:  1. https://github.com/beanstalkd/beanstalkd/blob/master/doc/protocol.txt#L133
4. `tube` - tube's name. Default: default.